plugin/src/main/java/org/opensearch/ml/action/models/GetModelTransportAction.java

```diff
@@ -77,37 +83,53 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLMode
         FetchSourceContext fetchSourceContext = getFetchSourceContext(mlModelGetRequest.isReturnContent());
         GetRequest getRequest = new GetRequest(ML_MODEL_INDEX).id(modelId).fetchSourceContext(fetchSourceContext);
         User user = RestActionUtils.getUserContext(client);
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
             ActionListener<MLModelGetResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
             client.get(getRequest, ActionListener.wrap(r -> {
                 if (r != null && r.isExists()) {
                     try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                        GetResponse getResponse = r;
-                        String algorithmName = getResponse.getSource().get(ALGORITHM_FIELD).toString();
-
+                        String algorithmName = r.getSource().get(ALGORITHM_FIELD).toString();
+                        Boolean isHidden = (Boolean) r.getSource().get(IS_HIDDEN_FIELD);
                         MLModel mlModel = MLModel.parse(parser, algorithmName);
-                        modelAccessControlHelper
-                            .validateModelGroupAccess(user, mlModel.getModelGroupId(), client, ActionListener.wrap(access -> {
-                                if (!access) {
-                                    wrappedListener
-                                        .onFailure(
-                                            new MLValidationException("User Doesn't have privilege to perform this operation on this model")
-                                        );
-                                } else {
-                                    log.debug("Completed Get Model Request, id:{}", modelId);
-                                    Connector connector = mlModel.getConnector();
-                                    if (connector != null) {
-                                        connector.removeCredential();
+                        if (isHidden != null && isHidden) {
+                            if (isSuperAdmin || !mlModelGetRequest.isUserInitiatedGetRequest()) {
```

Comment:
During that case we stash the context and send the request, so that we can pull information from index as super admin. During that time every request performs as a super admin request.

Reply:
And this is same for normal user request vs super admin request.

Reply:
I see, thanks.